src/shapes/group.class.js

```diff
@@ -385,6 +410,7 @@
      * @param {CanvasRenderingContext2D} ctx Context to render on
      */
     drawObject: function(ctx) {
+      this._renderBackground(ctx);
```

Comment:
mmmmm i really don't want this. can the group be a container only?

Reply:
Could you explain why you so against??
I really need/want it

Reply:
because a group should be a container, that handles layout. if you want a colored background you add a colored rect in this case.
But then....
can i have a gradient? can i have a picture?  can the background be a pattern?
And why there is no stroke?

This goes downhill fast.

How are you using it? that would clarify a lot.
In general background color for objects was a mistake imho.

Reply:
Well I have a solution.
The thing is I need to define an object that spreads over it's containing parent.
I can adapt the layout engine to check for objects with a flag that indicates that and it solves all issues I guess.
Thoughts?

Reply:
Same goes IMO for canvas' bg/overlay props

Reply:
yes they are, more the overlay image than the background.
Especially also the option they have to ignore the zoom.

But said so, can you explain your use case for group background?

Reply:
words! i have no idea what that drawing does and which is the background of the group and which is the group.
If if a secret, dump me an email, i just want to understand.

Reply:
canvas bg is the light blue color.
group bg is white.
You see objects (paths, rect, circle) that are part of groups.

Reply:
i don't know, let's keep it.
But there will be time in which i will tell you `I TOLD YOU`

Reply:
Alright I give in.
I trust you nd your experience and I can sense the downhill.
But can we agree on supporting `layout: 'fill-parent'` instead in a dedicated PR?

Reply:
I have found an elegant solution!
Layer is configured to fill parent at layout.
The solution is then:
1. wrap the object you want to act as bg/overlay with Layer 
2. listen to the layer's `layout` event or override the `onLayout` method (depends when you hook onto it)
3. change bg/overlay size from within the event!

https://user-images.githubusercontent.com/34343793/162664147-1282f3aa-c06a-453f-88c2-6c6470c5f313.mp4

Tell me if I can dump backgroundColor support

Reply:
Once Layer is merged we can open a PR to dump bg/overlay and perhaps add a bit of code in `loadFromJSON` that supports loading them and parsing them into a layer for backward comptibility.
Thoughts?

Reply:
I love this.
Thanks for challenging me not to accept `backgroundColor | fill | stroke`

Reply:
solved by #7901

Reply:
Didn't read all of this till now.
In my idea everything that is not the standard layout or the clipPath one will be reworked to be a plugin or a layout engine.
I'm not sure what a layout  is yet, but the core fabricJS should do the minimum possible. I ll get there on patch 3 or 4 i suppose